### PR TITLE
Refactoring of tests

### PR DESCRIPTION
**Pull request by phala**
Refactor and rewrite some tests. This will make the test suite sane.

### CODE REVIEW

File: nurseconnect/forms.py [hunk 1]
```diff
@@ -434,23 +427,39 @@ class ResetPasswordForm(forms.Form):
...
+            attrs={
+                "placeholder": _("Cellphone number"),
+                "class": "Form-input"
+            }
+        ),
+        label=_("Cellphone number")
+    )
 
-        return self.cleaned_data
+    # def clean(self):
```
Comment (chrislombaard) [inline nurseconnect/forms.py]: If you're happy with going with the above implementation, (I am), then please removed commented code.

File: nurseconnect/tests/test_profiles.py [hunk 2]
```diff
@@ -1,17 +1,121 @@
+"""
+Tests for the registration process.
+
+The registration process is broken down into three steps:
+1) MSISDN: for userame and password
+2) Security questions: getting answers to be used for password recovery
+3) Clinic code: For obtaining the user's clinic codde
```
Comment (chrislombaard) [inline nurseconnect/tests/test_profiles.py]: Minor issue: spelling error in "code"

File: nurseconnect/tests/test_profiles.py [hunk 3]
```diff
@@ -1,17 +1,121 @@
...
 from django.contrib.auth.models import User
 from django.test import TestCase
 from django.test.client import Client
 
+import mock
+from mock import patch
+
+from molo.core.models import SiteLanguage
 from molo.core.tests.base import MoloTestCaseMixin
+from molo.profiles.models import SecurityQuestion
```
Comment (chrislombaard) [inline nurseconnect/tests/test_profiles.py:20]: Space after `molo` imports.

File: nurseconnect/tests/test_profiles.py [hunk 4]
```diff
@@ -1,17 +1,121 @@
...
+    """
+    Tests for the 3 step registration process.
+    Successful flow through the steps is as follows
+    MSISDN -> Security Questions -> Clinic code
+    """
     def setUp(self):
+        self.mk_main()
         self.client = Client()
+
+        # site = Site.objects.get(is_default_site=True)
```
Comment (chrislombaard) [inline nurseconnect/tests/test_profiles.py]: Implement, or remove commented code.

File: nurseconnect/tests/test_profiles.py [hunk 5]
```diff
@@ -1,17 +1,121 @@
...
+        )
+        self.assertRedirects(
+            response, reverse("user_register_clinic_code")
+        )
+
+        # post clinic code step
+        response = self.client.post(
+            reverse("user_register_clinic_code"),
+            {
+                "clinic_code": "000000",
```
Comment (chrislombaard) [inline nurseconnect/tests/test_profiles.py:90]: How does a clinic code `000000` test as valid?

Comment (chrislombaard) [reply nurseconnect/tests/test_profiles.py:90]: Ah I see at the bottom you've mocked a clinic code using `000000`

File: nurseconnect/tests/test_views.py [hunk 6]
```diff
@@ -1,25 +1,60 @@
+from django.contrib.auth.models import User
 from django.core.urlresolvers import reverse
 from django.test import TestCase, Client
+
+from molo.core.models import SiteLanguage
 from molo.core.tests.base import MoloTestCaseMixin
-# from molo.profiles.models import SecurityQuestion
+from molo.profiles.models import SecurityQuestion
```
Comment (chrislombaard) [inline nurseconnect/tests/test_views.py:7]: Space after the `molo` imports.

File: nurseconnect/tests/test_profiles.py (hunk 3 above)
Comment (phala) [reply nurseconnect/tests/test_profiles.py:20]: Nope. They are the same type of imports. django libraries

**Pull request merged**